### DISCUSSION TITLE
Apollo: onError not firing on mutations

### DISCUSSION THREAD

**Discussion by dan-klasson** [dotansimha/graphql-code-generator#5773] [General] [answered]
I have this mutation:

```
  const [mutatePriceTagMutation] = useMutatePriceTagMutation({
    variables: {
      input: foo.getData(),
      id: foo.id,
    },
    onError: (error: ApolloError) => console.log(error),
  });

  const handleSave = async (event: MouseEvent<HTMLElement>) => {
    event.preventDefault();
    mutatePriceTagMutation();
  };
```

And it does not fire. `onError` on queries work just fine. This is a temporary workaround:

```
  const [mutatePriceTagMutation, { error }] = useMutatePriceTagMutation({
    variables: {
      input: foo.getData(),
      id: foo.id,
    },
  });
  if (error) console.log(error);
```

I'm using `"@apollo/react-hooks": "^3.1.4"`

**Comment by ardatan** [comment, marked as answer]
As you can see in the generated code by GraphQL Codegen, we only add type annotations as generics to existing Apollo React Hooks functions so your problem doesn't seem related to GraphQL Codegen. I'd suggest to create an issue on Apollo React repo.

**Reply by Benanna2019** [reply]
@ardatan would you say it is also an Apollo React issue if I am not being allowed to pass refetchQueries into my generated hooks? So I have a signin mutation hook and it does not seem to like refetchQueries.

**Reply by ardatan** [reply]
You can see what we generate. It just wraps their hooks with generics.
https://github.com/dotansimha/graphql-code-generator/blob/master/dev-test/githunt/types.reactApollo.tsx#L628
I think it is an issue on Apollo React side. Have you tried this with non-generated hooks?
